### PR TITLE
ss/narrowband-gain-error

### PR DESCRIPTION
**Pull request by ssiegelx**
Adds a new method `rfi.iterative_hpf_masking` that implements the algorithm used to identify narrowband features in the gains.  I am putting it in `ch_util` in case we want to use it in the real-time pipeline in the future.

Also makes a few modifications to `cal_utils.interpolate_gain` to make it work with the pipeline used to flag these features.

### CODE REVIEW

File: ch_util/rfi.py [hunk 1]
```diff
@@ -613,6 +613,171 @@ def nanmedian(*args, **kwargs):
         return np.nanmedian(*args, **kwargs)
 
 
+# Iterative HPF masking for identifying narrow-band features in gains or spectra
+def highpass_delay_filter(freq, tau_cut, flag, epsilon=1e-10):
+    """Construct a high-pass delay filter.
+
+    The stop band will range from [-tau_cut, tau_cut].
```
Comment (tristpinsm) [inline ch_util/rfi.py:620]: could we add a link to a reference for the HPF design? I was confused as to how this works until I realised it is like the DAYENU filter.

**Review by tristpinsm — COMMENTED** [review]
Just a couple things that caught my attention. Sorry for the delay!

File: ch_util/cal_utils.py [hunk 2]
```diff
@@ -2086,23 +2092,26 @@ def interpolate_gain(freq, gain, weight, flag=None, length_scale=30.0):
...
             ypred, err_ypred = gp.predict(xtest, return_std=True)
 
-            interp_gain[test, ii] = (ypred[:, 0] + ytrain_mu[:, 0]) + 1.0j * (
-                ypred[:, 1] + ytrain_mu[:, 1]
-            )
+            interp_gain[test, ii] = ypred[:, 0] + ytrain_mu[:, 0]
+            if iscomplex:
+                interp_gain[test, ii] += 1.0j * (ypred[:, 1] + ytrain_mu[:, 1])
+
             interp_weight[test, ii] = tools.invert_no_zero(err_ypred**2)
```
[the remark this answers is not in the document]
Comment (ssiegelx) [reply ch_util/cal_utils.py:2125]: Good catch.  In older versions of `sklearn` (<1.1.2), the `predict` method of the `GaussianProcessRegressor` returns a single error estimate even when there are multiple targets (despite what it says in the documentation).

However, in the most recent version a separate error estimate is returned for each target.  For the gain interpolation, the error estimates are equal for the real and imaginary components (and also equal to the single estimate returned in older versions).  I have updated the code to check if there is multiple estimates, and if so collapse over that axis by taking the average of the variance prior to saving to the weight dataset.  This should yield consistent behavior across versions.

**Comment by ssiegelx** [comment]
Thanks @tristpinsm!  I have addressed your comments.

**Review by tristpinsm — DISMISSED** [review]

**Review by tristpinsm — APPROVED** [review]

**Pull request merged**